src/array/ops/cast.rs

```diff
@@ -701,12 +705,120 @@ impl EmbeddingArray {
 
 impl ImageArray {
     pub fn cast(&self, dtype: &DataType) -> DaftResult<Series> {
-        self.physical.cast(dtype)
+        match dtype {
+            #[cfg(feature = "python")]
+            DataType::Python => {
+                Python::with_gil(|py| {
+                    let mut ndarrays = Vec::with_capacity(self.len());
+                    let mut bitmap = arrow2::bitmap::MutableBitmap::with_capacity(self.len());
+                    for i in 0..self.len() {
+                        let img_buf = self.as_image_obj(i);
+                        let ndarray = img_buf
+                            .map(|buf| {
+                                let slice = buf.as_u8_slice();
+                                let shape = (
+                                    buf.height() as usize,
+                                    buf.width() as usize,
+                                    buf.mode().num_channels() as usize,
+                                );
+                                // NOTE(Clark): This will create a copy in the Python heap, using NumPy for allocation.
+                                PyArray::from_slice(py, slice).reshape(shape.into_dimension())
+                            })
+                            .transpose()
```

Comment:
Why do we have to transpose here?

Reply:
To turn an `Option<Result<>>` into a `Result<Option<>>`.